File: frontend/src/lib/types/ai-connection.ts
export type AIProviderKey = "gemini" | (string & {});

export type AIConnectionStatus = {
    configured: boolean;
    provider?: AIProviderKey;
    model?: string;
    status?: "valid" | "invalid" | "error";
    lastValidatedAt?: string;
};

export type AIConnectionRequest = {
    provider: AIProviderKey;
    model: string;
    apiKey: string;
};

File: backend/app/schemas/ai_connection.py
from datetime import datetime

from pydantic import BaseModel


class AIConnectionCreate(BaseModel):
    provider: str
    model: str
    api_key: str


class AIConnectionStatus(BaseModel):
    configured: bool
    provider: str
    model: str
    status: str
    last_validated_at: datetime


class AIConnectionNotConfigured(BaseModel):
    configured: bool = False
